scripts/release/marketplaceRelease.js

```diff
@@ -0,0 +1,135 @@
+const nodefetch = require("node-fetch");
+const { join } = require("path");
+
+const config = {
+    appStoreUrl: "https://appstore.home.mendix.com/rest/packagesapi/v2"
+};
+
+main().catch(e => {
+    console.error(e);
+    process.exit(-1);
+});
+
+async function main() {
+    const {
+        widgetName,
+        version,
+        marketplace: { studioProMinimumVersion, marketplaceId }
+    } = require(join(process.cwd(), "package.json"));
+
+    if (!widgetName || !version || !studioProMinimumVersion || !marketplaceId || !version.includes(".")) {
+        throw new Error("Widget does not define expected keys in its package.json");
+    }
+
+    await uploadModuleToAppStore(widgetName, marketplaceId, version, studioProMinimumVersion);
+}
+
+async function uploadModuleToAppStore(widgetName, marketplaceId, version, studioProMinimumVersion) {
+    try {
+        const postResponse = await createDraft(marketplaceId, version, studioProMinimumVersion);
+        await publishDraft(postResponse.Version.VersionUUID);
+        console.log(`Successfully uploaded ${widgetName} to the Mendix Marketplace.`);
+    } catch (error) {
+        error.message = `Failed uploading module to appstore with error: ${error.message}`;
+        throw error;
+    }
+}
+
+async function getGithubAssetUrl() {
+    console.log("Retrieving informations from Github Tag");
+    const request = await nodefetch("https://api.github.com/repos/mendix/widgets-resources/releases?per_page=5", {
+        method: "GET"
+    });
+    const data = (await request.json()) ?? [];
+    const releaseId = data.filter(info => info.tag_name === process.env.TAG)?.pop()?.id;
+    if (releaseId) {
+        const assetsRequest = await nodefetch(
+            `https://api.github.com/repos/mendix/widgets-resources/releases/${releaseId}/assets`,
+            { method: "GET" }
+        );
+        const assetsData = (await assetsRequest.json()) ?? [];
+        const downloadUrl = assetsData.filter(asset => asset.name.endsWith(".mpk"))?.pop()?.browser_download_url;
+        if (!downloadUrl) {
+            throw new Error("Could not retrieve MPK url from GitHub tag");
+        }
+        return downloadUrl;
+    } else {
+        throw new Error("Could not retrieve release tag information from GitHub");
+    }
+}
+
+async function createDraft(marketplaceId, version, studioProMinimumVersion) {
+    console.log(`Creating draft in the Mendix Marketplace..`);
+    const url = `${config.appStoreUrl}/packages/${marketplaceId}/version`;
+    try {
+        const [major, minor, patch] = version.split(".");
```

Comment:
I think it's wise to check whether the version consists of 3 number, otherwise throw an error.

Reply:
Fixed